### PR TITLE
Update scope validation using XACML policies

### PR DESCRIPTION
**Pull request by ShanChathusanda93**
Fixes : https://github.com/wso2/product-is/issues/18801
Ref : https://is.docs.wso2.com/en/5.9.0/learn/validating-the-scope-of-oauth-access-tokens-using-xacml-policies/#validating-the-scope-of-oauth-access-tokens-using-xacml-policies

### CODE REVIEW

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 1]
```diff
@@ -1,6 +1,197 @@
...
+Server so that the authentication happens as expected.
+
+1. On the {{ product_name }} Console, go to **Applications**.
+2. Click **New Application** and select **Standard-Based Application**.
+   ![Register a standard-based application]({{base_path}}/assets/img/guides/applications/register-an-sba.png){: width="600" style="display: block; margin: 0 auto; border: 0.3px solid lightgrey;"}
+3. Provide an application name and select the other options based on your requirements.
+
+    !!! note 
+        - You can choose OIDC or SAML as the standard protocol for your application. See the complete list of [OIDC]({{base_path}}/references/app-settings/oidc-settings-for-app/) and [SAML]({{base_path}}/references/app-settings/saml-settings-for-app/)  configurations.
+        - If you use OIDC, you can configure a management app, which can access the management APIs in {{ product_name }}. Learn about [invoking management APIs]({{base_path}}/apis/{{ api_authentication_path }}/).
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:28]: There is no management app concept in IS-7.0.0

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 2]
```diff
@@ -1,6 +1,197 @@
...
+9. Execute the Inbound Protocols PUT REST API call to update the application with the 
+   XACML scope validator which added in the step 7. You can get the API call details from
+   <a href="{{base_path}}/apis/application-rest-api/#tag/Inbound-Protocols-OAuth-OIDC/operation/updateInboundOAuthConfiguration/">Update OIDC authentication protocol parameters</a>.
+
+### Register an API resource and Authorize for an app
+
+1. Go to the **API Resources** and click **New API Resource**.
+2. Add the relevant details to Basic Details, Scopes, Authorization sections and click **Create**.
+3. Go to the **API Authorization** tab in the created application.
+4. Click **Authorize an API Resource**, select the created API resource, scope and click **Finish**.
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:59]:
```suggestion
4. Click **Authorize an API Resource**, select the created API resource, scopes and click **Finish**.
```

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 3]
```diff
@@ -1,6 +1,197 @@
...
+
+    !!!note
+        For more information on API Authorization, see [here](../api-authorization/api-authorization.md)
+
+### Create a user and a role
+
+1. Go to the **User Management** and click **Users**.
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:66]:
```suggestion
1. Go to **User Management** > **Users**
```
keep consistency

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 4]
```diff
@@ -1,6 +1,197 @@
...
+2. Click **Add User** button and select **Single User** option, provide the relevant details and click 
+   **Save & Continue**.
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:68]: break into two steps

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 5]
```diff
@@ -1,6 +1,197 @@
...
+3. Go to the **User Management** and click **Roles**.
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:69]:
```suggestion
3. Go to **User Management** > **Roles**.
```

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 6]
```diff
@@ -1,6 +1,197 @@
...
+7. On the UI that appears, leave the default values as they are and
+   click **Publish**.
+
+   To ensure that the policy has been published successfully, click on
+   **Policy View** under the **Entitlement \> PDP** section on the
+   **Main** tab of the Management Console, and check if the created
+   policy is listed.
+
+Now that you have created the policy and enforced it using the policy
+template, you can test the policy to evaluate whether XACML scope is
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:113]:
```suggestion
template. You can test the policy to evaluate whether XACML scope is
```

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 7]
```diff
@@ -1,6 +1,197 @@
...
+3. Execute the following cURL command to get an access token using the password grant type.
+
+```java
+curl --location 'https://localhost:9443/t/<TENANT_DOMAIN>/oauth2/token' \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--header 'Authorization: Basic <BASE64_ENCODED(CLIENT_ID:CLIENT_SECRET)>' \
+--data-urlencode 'grant_type=password' \
+--data-urlencode 'username=<USER_NAME>' \
+--data-urlencode 'password=<USER_PASSWORD>' \
+--data-urlencode 'scope=openid SCOPE_1'
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:131]:
```suggestion
--data-urlencode 'scope=openid <SCOPE_1> <SCOPE_2>'
```

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 8]
```diff
@@ -1,6 +1,197 @@
...
+```
+If the token request can be permitted by the policy, then the response will contain an access token.
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:133]:
```suggestion
If the token request can be permitted by the policy, the response will contain an access token. 
```

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 9]
```diff
@@ -1,6 +1,197 @@
...
+curl --location 'https://localhost:9443/t/<TENANT_DOMAIN>/oauth2/token' \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--header 'Authorization: Basic <BASE64_ENCODED(CLIENT_ID:CLIENT_SECRET)>' \
+--data-urlencode 'grant_type=password' \
+--data-urlencode 'username=<USER_NAME>' \
+--data-urlencode 'password=<USER_PASSWORD>' \
+--data-urlencode 'scope=openid SCOPE_1'
+```
+If the token request can be permitted by the policy, then the response will contain an access token. 
+If the token request is denied by the policy, then the response will contain an error message.
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:134]:
```suggestion
If the token request is denied by the policy, the response will contain an error message.
```

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md [hunk 10]
```diff
@@ -1,6 +1,197 @@
...
+
+Follow the steps below to try out the policy using the XACML TryIt tool.
+
+!!! tip
+    The XACML TryIt tool allows you to test policies easily without having
+    to create and send authorization requests to WSO2 IS. It is a tool
+    through which authorization requests can be created and evaluated
+    against available policies. You can write simple XACML 3.0 requests in
+    XML format and try them using the web UI of the TryIt tool
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:143]:
```suggestion
    XML format and try them using the web UI of the TryIt tool.
```

File: en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md (hunk 7 above)
Comment (ShanChathusanda93) [reply en/identity-server/next/docs/guides/authorization/api-authorization/attribute-based-access-control.md:131]: SCOPE_1 is the actual scope name that we need to validate from the XACML policy

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**